test/test_transforms.py

```diff
@@ -1315,6 +1196,130 @@ def test_random_erasing(self):
         t.__repr__()
 
 
+def test_convert_image_dtype_float_to_float(self):
+    for input_dtype, output_dtypes in cycle_over(float_dtypes()):
+        input_image = torch.tensor((0.0, 1.0), dtype=input_dtype)
+        for output_dtype in output_dtypes:
```

Comment:
Thanks for the PR @artifiitr !

We should remove `self` from the function and also try to parametrize over `input_dtype` and `output_dtype`. This way we should be able to get rid of the self.subTest as well. You can find a complete example of parametrization here https://github.com/pytorch/vision/pull/3907/files.

Reply:
@NicolasHug , In 1200th line, output_dtype**s** contains only one value acccording to cycle_over(float_dtypes), So can I eliminate output_dtype**s** from above function and use only output_dtype?

Reply:
From what I can tell this yields 2 elements, not just one:

```
(torch.float32, (torch.float64,))
(torch.float64, (torch.float32,))
```

let's just change this into a parametrization like

```py
@pytest.mark.parametrize('input_dtype, output_dtypes', cycle_over(float_dtypes()))
def test_blahblah(input_dtype, output_dtypes):
```

Reply:
So shall I keep the for loop in 1202 line?

Reply:
no, the for loops should be repalced with parametrization. You can refer to the porting guide I wrote in https://github.com/pytorch/vision/issues/3945#issue-909493806, there are some illustration links

Reply:
@Nicholas, I think he is trying to say that
```
for output_dtype in output_dtypes:
```
in the above line , output_dtypes is a list of only 1 item. So aren't output_dtype and output_dtypes same in above function? And we can parametrize one out of them only.

Reply:
@NicolasHug Outer loop can be parametrized as u said. But inner loop is dependent on outer one. In the porting guide, I couldn't find parameterization of such loop. So, how should I parametrize it?

Reply:
Sorry for the confusion @artifiitr , indeed here we can parametrize over  

```py
@pytest.mark.parametrize('input_dtype, output_dtypes', cycle_over(float_dtypes()))
def test_blahblah(input_dtype, output_dtypes):
```
but it seems that we'll need to keep the inner loop:

```py
for output_dtype in output_dtypes:
    ...
```